cli/azd/pkg/pipeline/github_provider.go

```diff
@@ -627,18 +619,100 @@ func (p *GitHubCiProvider) configurePipeline(
 	provisioningProvider provisioning.Options,
 	additionalSecrets map[string]string,
 	additionalVariables map[string]string,
+	options *ConfigOptions,
 ) (CiPipeline, error) {
-
 	repoSlug := repoDetails.owner + "/" + repoDetails.repoName
+
+	// Variables and Secrets for a gh-actions are independent from the gh-action. They are set on the repository level.
+	// We need to clean up the previous values before setting the new ones.
+	// By doing this, we are handling:
+	// - When a secret is moved to be a variable (or vice versa). Don't leak the previous value on the pipeline.
+	// - When there was a previous additional variable/secret set and then it was updated to empty string or unset from .env.
+	msg := ""
+	var procErr error
+	ciSecrets, ciVariables := []string{}, []string{}
```

Comment:
Are we able to include some new unit tests here to validate this expected behavior?

Reply:
I think I can add coverage here. Yes.
I'll work on it